client/tinychain/ml/dnn.py

```diff
@@ -49,13 +135,49 @@ def neural_net(layers):
 
     class DNN(NeuralNet):
         def eval(self, inputs):
-            state = layers[0].eval(inputs)
+            '''
+            eval function it will help you to get the result
+            of each layer
+            
+            for the first layer its exception we eval the inputs
+            
+            type(inputs) : tc.tensor.Dense
+            state : is the Result from evaluating  cascaded layers 
+            '''
+            state = layers[0].eval(inputs) 
             for i in range(1, len(layers)):
                 state = layers[i].eval(state)
 
             return state
+       
+        def adam_init(self):
```

Comment:
This should be part of the optimizer class, not DNN. It may be used for many other architectures, complex neural nets made from multiple simple architectures, etc